### PR TITLE
feat: pointer-capture API supported

### PR DESCRIPTION
**Pull request by supminn**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)
- #131 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update. README.md has been updated.

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. 

# Checklist:

- [x] My code follows the [style guidelines](https://github.com/atapas/webapis-playground/blob/master/HOW-TO-ADD-DEMO.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# Screen recording of local testing.

https://user-images.githubusercontent.com/30731236/194105127-82dfb95b-5200-44fa-a39d-a3098565b4e4.mp4

### CODE REVIEW

**Comment by supminn** [comment]
Hi @atapas 
I have added an implementation of my understanding. Please review and let me know your feedback.

**Comment by atapas** [comment]
@supminn What's the expectation when we turn it OFF?

**Comment by supminn** [comment]
> @supminn What's the expectation when we turn it OFF?

If we try to drag the red dot (timeline slider) when the `pointer-capture` is turned off, the slider would stop moving if the cursor moves out of the focus area (i.e. when the cursor is moved away from the red dot while dragging).
Whereas, when the `pointer-capture` is turned on since the `pointerId` is captured; the slider keeps moving even when the cursor is being dragged out of the focus area.


I have tried to add a better explanation to the preview. Do let me know if it could be rephrased in an easier manner.

**Comment by atapas** [comment]
> > @supminn What's the expectation when we turn it OFF?
> 
> If we try to drag the red dot (timeline slider) when the `pointer-capture` is turned off, the slider would stop moving if the cursor moves out of the focus area (i.e. when the cursor is moved away from the red dot while dragging). Whereas, when the `pointer-capture` is turned on since the `pointerId` is captured; the slider keeps moving even when the cursor is being dragged out of the focus area.
> 
> I have tried to add a better explanation to the preview. Do let me know if it could be rephrased in an easier manner.

I'm still a bit confused. Can we have a call tomorrow, @supminn ?

**Comment by supminn** [comment]
> > > @supminn What's the expectation when we turn it OFF?
> > 
> > 
> > If we try to drag the red dot (timeline slider) when the `pointer-capture` is turned off, the slider would stop moving if the cursor moves out of the focus area (i.e. when the cursor is moved away from the red dot while dragging). Whereas, when the `pointer-capture` is turned on since the `pointerId` is captured; the slider keeps moving even when the cursor is being dragged out of the focus area.
> > I have tried to add a better explanation to the preview. Do let me know if it could be rephrased in an easier manner.
> 
> I'm still a bit confused. Can we have a call tomorrow, @supminn ?

sure @atapas, Shall I reach out to you on twitter or discord to take this forward?

File: src/modules/apis/pointer-capture/index.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+export const hasSupport = (): boolean => {
+  return true;
```
Comment (atapas) [inline src/modules/apis/pointer-capture/index.ts:3]: @supminn is it supported by all the browsers?

Comment (supminn) [reply src/modules/apis/pointer-capture/index.ts]: I have updated the implementation for `hasSupport`.

**Review by atapas — APPROVED** [review]
@supminn Great, thanks for putting it together. Accepted.

**Comment by atapas** [comment]
Good Job, Don't forget to let the world know about your contribution :)

**Comment by atapas** [comment]
@all-contributors please add @supminn for Code

**Pull request merged**